### ISSUE TITLE
Enable debugging or timeout

### ISSUE THREAD

**Issue by jstolp** [paratestphp/paratest#317]
Hello,

First of all... I <3 the lib! great work.
Is there any way in order to log the current test/worker failing, we are getting some deadlock on the database (good so we know and can fix it) but it's hard to identify which test is currently locking the db.

Can i enable some debugging or timeout via PHPUnit or the WrapperRunner config.

Do you have other tips, or things i could't try... Else it would be a great feature to use

```--debug``` or ``` -vvv ``` or something as a feature.

**Comment by michaelbutler** [comment]
Yes you may implement a test listener which logs to a file, you can log on start and stop and then find which never log for stop. Using xdebug could be tough with so many processes happening in parallel.

**Comment by jstolp** [comment]
@michaelbutler thanks man, i was on the same idea, just checking. If i have something valuable or prod-ish ready i'll share it. (Also for future references this topic could be handy)

**Comment by tarjei** [comment]
Hi, I just had to debug an error that was only happening within paratest, here's what I did. The instructions might not be complete, but I think the gist of them are.


1. I started by managing to reproduce the error with paratest running with a single process:
```
php7.4 bin/paratest -v4 --runner WrapperRunner --configuration ./phpunit-t.0_1_2.xml --phpunit=bin/phpunit -p1
```

The configuration file referenced before is one I generated by doing a binary search over all combinations of tests create the smallest testset possible for the tests to fail - on my case just two tests.

2. Xdebug must be installed and working. I added `xdebug.log="/tmp/xdebug.log"` to my config but apart form that nothing. Also, I had to change the portnumber in PHPStorm to 9003 from the default 9000. PHPStorm configuration is a bit outside the scope of this comment but you might have to fiddle a bit to get it working. 

3. The paraunit process uses WrapperRunner to create a set of WrapperWorkers. Each of the WrapperWorkers extend BaseWorker. On startup each WrapperWorker will create a subprocess that calls `phpunit_wrapper` and communicate with it using stdin. It is this subprocess that you want to debug.

In BaseWorker I found the spot where the environment variables to the phpunit_wrapper process got set and added the ones I needed: 
```Runners\PhpUnit\Worker\BaseWorker.php 
// around line 34:
$bin .= 'XDEBUG_SESSION=1  XDEBUG_MODE=debug ';
```

4. Then I opened up the phpunit_wrapper in PHPStorm and set a breakpoint. 

5. Then I ran paratest on the commandline: 
```
php7.4 bin/paratest -v4 --runner WrapperRunner --configuration ./phpunit-t.0_1_2.xml --phpunit=bin/phpunit -p1
```

Now I had a debug session going.
